File: 00-node/01-examples/04-get-and-exception-handling/service/book.js
'use strict'

let book = {
    getAllBooks: (req, res) => {
        let books = [
            {title: 'Code Complete 2nd Edition', author: 'Steve McConnell', year: '2004'}, 
            {title: 'Clean Code', author: 'Robert C. Martin', year: '2008'}, 
            {title: 'The Last Lecture', author: 'Randy Pausch', year: '2008'} 
        ];

        throw new Error('Error here!');

        res.send(books);
    }
};

module.exports = book;

File: 02-netcore/01-examples/05-get-and-unit-test/src/Core/Book.cs
using System.Collections.Generic;
using _05_get_and_unit_test.Core.Models;

namespace _05_get_and_unit_test.Core
{
    public class Book
    {
        public IEnumerable<BookInformation> GetAllBooks()
        {
            var books = new List<BookInformation>();
            books.Add(new BookInformation("Code Complete 2nd Edition", "Steve McConnell","2004"));
            books.Add(new BookInformation("Clean Code", "Robert C. Martin","2008"));
            books.Add(new BookInformation("The Last Lecture", "Randy Pausch","2008"));
            
            return books.ToArray();
        }

    }
}